sdk/azidentity/aad_identity_client.go

```diff
@@ -373,3 +381,65 @@ func (c *aadIdentityClient) createDeviceCodeNumberRequest(ctx context.Context, t
 	}
 	return req, nil
 }
+
+// authenticateInteractiveBrowser opens an interactive browser window, gets the authorization code and requests an Access Token with the
+// authorization code and returns the token or an error in case of authentication failure.
+// ctx: The current request context
+// tenantID: The Azure Active Directory tenant (directory) ID of the service principal
+// clientID: The client (application) ID of the service principal
+// clientSecret: Gets the client secret that was generated for the App Registration used to authenticate the client.
+// scopes: The scopes required for the token
+func (c *aadIdentityClient) authenticateInteractiveBrowser(ctx context.Context, tenantID string, clientID string, clientSecret string, redirectURI string, scopes []string) (*azcore.AccessToken, error) {
+	cfg, err := authCodeReceiver(c.options.AuthorityHost, tenantID, clientID, redirectURI, scopes)
+	if err != nil {
+		return nil, err
+	}
+	return c.authenticateAuthCode(ctx, tenantID, clientID, cfg.authCode, clientSecret, cfg.redirectURI, scopes)
+}
+
+// authenticateAuthCode requests an Access Token with the authorization code and returns the token or an error in case of authentication failure.
+// ctx: The current request context
+// tenantID: The Azure Active Directory tenant (directory) ID of the service principal
+// clientID: The client (application) ID of the service principal
+// clientSecret: Gets the client secret that was generated for the App Registration used to authenticate the client.
+// scopes: The scopes required for the token
+func (c *aadIdentityClient) authenticateAuthCode(ctx context.Context, tenantID string, clientID string, authCode string, clientSecret string, redirectURI string, scopes []string) (*azcore.AccessToken, error) {
```

Comment:
Separating this because I'm adding an AuthorizationCodeCredential once this is merged.